drop-in/src/main/java/com/adyen/checkout/dropin/internal/provider/ComponentParsingProvider.kt

```diff
@@ -388,7 +388,8 @@ internal fun getPaymentMethodAvailabilityCheck(
     sessionDetails: SessionDetails?,
 ): PaymentMethodAvailabilityCheck<Configuration> {
     val dropInParams = dropInConfiguration.mapToParams(amount)
-    val sessionParams = sessionDetails?.let { SessionParamsFactory.create(it) }
+    // TODO: Once Backend provides the correct amount in the SessionSetupResponse use that in SessionDetails
+    val sessionParams = sessionDetails?.let { SessionParamsFactory.create(it.copy(amount = amount)) }
```

Comment:
Maybe let's move this logic to a common helper/extension function to make it easy to change it later